src/app_level/player.c

```diff
@@ -0,0 +1,2997 @@
+            if (DashEngineSoundTimeout == 0) {
+                playDashEngineSounds(TRUE);
+            }
+        } else {
+            playDashEngineSounds(FALSE);
+            IsDashEngineOn = 0;
+            Icons_SetDashEngineEnabled(FALSE);
+        }
+    }
+    if (DashEngineSoundTimeout > 0) {
+        DashEngineSoundTimeout--;
+    }
+    TotalSpeedMult = DashEngineSpeedMult * 0.0005f * (1.0f + D_803AE478_54E888 * 0.25f);
+}
+
+void handleCButtons(GObj* obj) {
+    s32 CButtonPressed = FALSE;
+
+    if (gDirectionIndex < 0) {
+        return;
+    }
+
+    if (!IsInputDisabled) {
+        if (gContInputPressedButtons & U_CBUTTONS) {
+            auPlaySound(SOUND_ID_4);
+            if (gDirectionIndex == 0) {
+                TurnToDirSpeed = 0.2f;
+            } else if (gDirectionIndex == 1) {
+                TurnToDirSpeed = 0.1f;
+            } else {
+                TurnToDirSpeed = -0.1f;
+            }
+            gDirectionIndex = 2;
+            CButtonPressed = TRUE;
+            ReticleScreenX = 0;
+        } else if (gContInputPressedButtons & R_CBUTTONS) {
+            auPlaySound(SOUND_ID_4);
+            gDirectionIndex++;
+            if (gDirectionIndex >= 4) {
+                gDirectionIndex -= 4;
+            }
+            CButtonPressed = TRUE;
+            TurnToDirSpeed = 0.1f;
+            ReticleScreenX = 0;
+        } else if (gContInputPressedButtons & L_CBUTTONS) {
+            auPlaySound(SOUND_ID_4);
+            gDirectionIndex--;
+            if (gDirectionIndex < 0) {
+                gDirectionIndex += 4;
+            }
+            CButtonPressed = TRUE;
+            TurnToDirSpeed = -0.1f;
+            ReticleScreenX = 0;
+        }
+    }
+
+    updateDashEngine();
+    if (CButtonPressed) {
+        StickXAccum = 0.0f;
+        PlayerViewYaw = 0.0f;
+        ohResumeProcessByFunction(obj, processTurnToDirection);
+        ohPauseProcessByFunction(obj, updateMovementState);
+        ohPauseProcessByFunction(obj, handleAnalogStick);
+        ohPauseProcessByFunction(obj, handleCButtons);
+        return;
+    }
+
+    if (ABS(ReticleScreenX) > 50) {
+        PlayerViewYaw += StickXValue * 0.03 * 0.8;
+    }
+}
+
+void processTurnToDirection(GObj* obj) {
+    f32 currentYaw;
+    f32 targetYaw;
+    s32 v0 = 0;
+
+    if (IsPaused) {
+        return;
+    }
+
+    targetYaw = DirectionsList[gDirectionIndex];
+    currentYaw = CurrentDirectionYaw;
+
+    while (targetYaw < 0.0f) {
+        targetYaw += TAU;
+    }
+    while (targetYaw >= TAU) {
+        targetYaw -= TAU;
+    }
+
+    while (currentYaw < 0.0f) {
+        currentYaw += TAU;
+    }
+    while (currentYaw >= TAU) {
+        currentYaw -= TAU;
+    }
+
+    if (ABS(targetYaw - currentYaw) > ABS(TurnToDirSpeed)) {
+        CurrentDirectionYaw += TurnToDirSpeed;
+    } else {
+        CurrentDirectionYaw = DirectionsList[gDirectionIndex];
+        v0++;
+    }
+    targetYaw = 0.04712389f; // need var reuse for matching
```

Comment:
3 pi / 200

Reply:
(I use wolframalpha for these 😁 https://www.wolframalpha.com/input?i=0.04712389 )